scripts/src/main/scala/com/gu/mediaservice/scripts/EsScript.scala

```diff
@@ -42,81 +49,132 @@ object MoveIndex extends EsScript {
+    val scrollSize = 500
+    val currentIndex = EsClient.currentIndex
+    val newIndex = EsClient.nextIndex
+
+    val from = if(args.isEmpty) None else Some(DateTime.parse(args(0)))
+    validateCurrentState(EsClient, from)
+    Await.result(reindex(from, EsClient), Duration.Inf)
+
+    def reindex(from: Option[DateTime], esClient: ElasticSearchClient) : Future[SearchResponse] = {
+      def _scroll(scroll: SearchResponse, done: Long = 0): Future[SearchResponse] = {
+        val client = esClient.client
+        val doing = done + scrollSize
+        System.out.println(scrollPercentage(scroll, doing, done))
+
+        def bulkFromHits(hits: Array[SearchHit]): BulkRequestBuilder = {
+          val bulkRequests : Array[IndexRequestBuilder] = hits.map { hit =>
+            client.prepareIndex(newIndex, imageType, hit.id).setSource(hit.source) }
+
+          val bulk = client.prepareBulk
+          bulkRequests map { bulk.add }
+          bulk
+        }
+        def scrollPercentage(scroll: SearchResponse, doing: Long, done: Long): String = {
           val total = scroll.getHits.totalHits
-          val doing = done + scrollSize
-
           // roughly accurate as we're using done, which is relative to scrollSize, rather than the actual number of docs in the new index
           val percentage = (done.toFloat / total) * 100
+          s"Reindexing $doing of $total ($percentage%)"
+        }
 
-          val hits = scroll.getHits.hits
-
-          if (hits.nonEmpty) {
-            // TODO: Abstract out logging
-            System.out.println(s"Reindexing $doing of $total ($percentage%)")
-
-            val bulk = client.prepareBulk
 
-            hits.foreach { hit =>
-              bulk.add(client.prepareIndex(newIndex, imageType, hit.id)
-                .setSource(hit.source))
-            }
+        val hits = scroll.getHits.hits
+        if(hits.nonEmpty) {
+          bulkFromHits(hits).execute.actionGet
+          val scrollResponse = client.prepareSearchScroll(scroll.getScrollId).setScroll(scrollTime).execute.actionGet
+          _scroll(scrollResponse, doing)
+        } else {
+          println("No more results found")
+          Future.successful[SearchResponse](scroll)
+        }
+      }
 
-            bulk.execute.actionGet
-            reindexScroll(client.prepareSearchScroll(scroll.getScrollId)
-              .setScroll(scrollTime).execute.actionGet, doing)
-          } else {
+      def query(from: Option[DateTime]) : SearchRequestBuilder = {
+        val queryType = from.map(time =>
+          rangeQuery("lastModified").from(from.get).to(DateTime.now)
+        ).getOrElse(
+          matchAllQuery()
+        )
 
-          }
-        }
+        EsClient.client.prepareSearch(currentIndex)
+          .setTypes(imageType)
+          .setScroll(scrollTime)
+          .setSize(scrollSize)
+          .setQuery(queryType)
+      }
 
-        Thread.sleep(1000)
 
-        reindexScroll(query.execute.actionGet)
+      if(from.isEmpty) {
```

Comment:
it's not clear! maybe i should throw it in a nicely named method. 

If the user supplies a "from" time (or it's been called recursively) then don't create a new index to populate. Basically, if you're asking for a partial update, then you don't want a new index.